src/services/housing.js

```diff
@@ -46,7 +49,7 @@ import './user';
  * @property {*} DateSubmitted The date the application was submitted, or null if not yet submitted
  * @property {*} DateModified The date the application was last modified
  * @property {String} Username Username of the primary applicant
- * @property {String[]} Applicants Array of student usernames
+ * @property {ApplicantPartialInfo[]} Applicants Array of objects with applicant username and off-campus program
```

Comment:
Should this object be named "Applicants" while your list of ApartmentApplicant objects in saveApartmentApplication is also named "applicants"?

Reply:
Yes. It is meant to match the variable named in the backend view model and the database because that typedef is used for the data returned by `getApartmentApplication` (the "load" endpoint, which is not yet implemented).
These typedefs exist mostly for documentation purposes, as I don't think the javascript does anything this enforce these variable typings, but I know it is parsed by the VSCode stuff to show the info in tooltips and it shows developers what properties they can expect to access from the data that is retrieved from the back.

This `Applicants` array is also meant to be the same as the `Applicants` array created within the `applicationDetails` object in `saveApartmentApplication`, whereas the lowercase `applicants` is a parameter of the function.

Reply:
Simply put, the `ApartmentApplicant` object type is used by the frontend because it is best to hold onto the full StudentProfileInfo object for each applicant 

The `ApplicantPartialInfo` object type is needed because we need to replace the StudentProfileInfo objects with just the usernames when we want to send/recieve data to/from the database

Reply:
That makes sense. Thanks, Christian